config/reader.go

```diff
@@ -39,7 +40,22 @@ func getAgentInfo() (*apppb.AgentInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	platform := fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH)
+
+	arch := runtime.GOARCH
+	// "arm" is used for arm32. "arm64" is used for versions after v7
+	if arch == "arm" {
+		// armv7 added LPAE (Large Page Address Extension).
+		// this is an official way to detect armv7
+		// https://go-review.googlesource.com/c/go/+/525637/2/src/internal/cpu/cpu_arm.go#36
+		if cpu.ARM.HasLPAE {
+			arch = "arm32v7"
+		} else {
+			// fallback to armv6
+			arch = "arm32v6"
```

Comment:
Question to reviewers:
- for versions of arm before armv6 (which will probably never be supported), is it better to fallback to v6 or some invalid string like "arm"?

Reply:
'arm' is the golang environment label for all arm32 so might be confusing?

your call, I would use same scheme you've been using (arm32v5)

Reply:
If we find a customer who has an arm32v5 board that we need to support, I will be on it to fix this. Until then, falling back to a working platform seems less error prone. I might be wrong about this... but we will see.